tests/health_events_analyzer_test.go

```diff
@@ -0,0 +1,122 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package tests
+
+import (
+	"context"
+	"testing"
+	"tests/helpers"
+
+	pb "github.com/nvidia/nvsentinel/data-models/pkg/protos"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+const (
+	ERRORCODE_13 = "13"
+	ERRORCODE_48 = "48"
+	ERRORCODE_31 = "31"
+)
+
+func TestMultipleRemediationsCompleted(t *testing.T) {
+	feature := features.New("TestMultipleRemediationsCompleted").
+		WithLabel("suite", "health-event-analyzer")
+
+	var testCtx *helpers.HealthEventsAnalyzerTestContext
+
+	feature.Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		var newCtx context.Context
+		newCtx, testCtx = helpers.SetupHealthEventsAnalyzerTest(ctx, t, c, "data/health-events-analyzer-config.yaml", "health-events-analyzer-test")
+
+		client, err := c.NewClient()
+		require.NoError(t, err)
+		helpers.TriggerMultipleRemediationsCycle(ctx, t, client, testCtx.NodeName)
+
+		return newCtx
+	})
+
+	feature.Assess("Check if MultipleRemediations node condition is added", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		client, err := c.NewClient()
+		assert.NoError(t, err, "failed to create client")
+		gpuNodeName := testCtx.NodeName
+
+		event := helpers.NewHealthEvent(gpuNodeName).
+			WithErrorCode(ERRORCODE_31).
+			WithRecommendedAction(int(pb.RecommendedAction_RESTART_VM))
+		helpers.SendHealthEvent(ctx, t, event)
+
+		helpers.WaitForNodeConditionWithCheckName(ctx, t, client, gpuNodeName, "MultipleRemediations", "ErrorCode:31 GPU:0 Recommended Action=CONTACT_SUPPORT;")
+
+		return ctx
+	})
+
+	feature.Teardown(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		return helpers.TeardownHealthEventsAnalyzer(ctx, t, c, testCtx.NodeName, testCtx.ConfigMapBackup)
+	})
+
+	testEnv.Test(t, feature.Feature())
+}
+
+func TestMultipleRemediationsNotTriggered(t *testing.T) {
+	feature := features.New("TestMultipleRemediationsNotTriggered").
+		WithLabel("suite", "health-event-analyzer")
+
+	var testCtx *helpers.HealthEventsAnalyzerTestContext
+
+	feature.Setup(func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
+		var newCtx context.Context
+		newCtx, testCtx = helpers.SetupHealthEventsAnalyzerTest(ctx, t, c, "data/health-events-analyzer-config.yaml", "health-events-analyzer-test")
+
+		gpuNodeName := testCtx.NodeName
+
+		t.Logf("Injecting non-fatal events to node %s", gpuNodeName)
+		for i := 0; i < 5; i++ {
+			event := helpers.NewHealthEvent(gpuNodeName).
+				WithFatal(false).
+				WithErrorCode(ERRORCODE_13).
+				WithRecommendedAction(int(pb.RecommendedAction_RESTART_VM))
+
+			helpers.SendHealthEvent(ctx, t, event)
+
+			helpers.SendHealthyEvent(ctx, t, gpuNodeName)
```

Comment:
Do we need to send healthy event after every unhealthy event? Should this be moved outside of the loop? Or feature.Teardown?